samples/dotnet/KernelBuilder/Program.cs

```diff
@@ -104,15 +104,33 @@
 // AI requests (when using the kernel).
 
 var kernel8 = Kernel.Builder
-    .Configure(c => c.SetRetryMechanism(new RetryThreeTimes()))
+    .Configure(c => c.SetHttpRetryHandlerFactory(new RetryThreeTimesFactory()))
     .Build();
 
-public class RetryThreeTimes : IRetryMechanism
+public class RetryThreeTimesFactory : IDelegatingHandlerFactory
 {
-    public Task ExecuteWithRetryAsync(Func<Task> action, ILogger log, CancellationToken cancellationToken = default)
+    public DelegatingHandler Create(ILogger log)
     {
-        var policy = GetPolicy(log);
-        return policy.ExecuteAsync((_) => action(), cancellationToken);
+        return new RetryThreeTimes(log);
+    }
+}
+
+public class RetryThreeTimes : DelegatingHandler
```

Comment:
Do you see any benefit of having a base class RetryNTimes? It would abstract out the logging and bit and make the class more flexible for customized retry policies. 

Or general logic and factories of RetryN times with 3 being a paramter. I get that it gets a little sketchy if someone codes >5 though.

Reply:
I'll add a comment here (and kernel-syntax-examples) -- but these are all meant to be examples of how folks would supply a custom handler. Maybe now that there is better built-in support, we can remove this from KernelBuilder sample -- cc @dluc